src/problems/cvrp/local_search/local_search.cpp

```diff
@@ -348,7 +349,7 @@ void cvrp_local_search::update_amounts(index_t v) {
   std::transform(_sol_state.fwd_amounts[v].cbegin(),
                  _sol_state.fwd_amounts[v].cend(),
                  _sol_state.bwd_amounts[v].begin(),
-                 [&](const auto& a) {
+                 [&](const auto& a) -> amount_t {
```

Comment:
I guess this is necessary so the compiler can resolve the return type, due to the template change for `amount_t`?

Reply:
No, it's actually a bit worse. With this annotation the return type of this lambda is `amount_t` and an implicit conversion at the return-statements converts the `amount_diff_t` into an `amount_t` by actually copying the values and everything is fine.

Without it the return type is inferred to be `amount_diff_t` and the return-statement does not perform any implicit conversions, this is a problem since `amount_diff_t` contains references to the lhs and rhs subexpressions. This way however undefined behaviour happens somewhere and this calculation is not carried out correctly. However now that I think about it, I understand the exact reason: the `auto total_amount = ` line copies into a local amount object. The lifetime of this local object ends after the return, but the `amount_diff_t` still holds a reference to it. I pushed a commit that fixes this issue a bit more elegantly.

This is the part I don't like about this PR, it makes it a bit harder to use amount_t correctly.